### PR TITLE
[flink] add CreateTagFromWatermarkAction

### PR DESCRIPTION
**Pull request by herefree**
### Purpose

add CreateTagFromWatermarkAction for old flink version

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/procedure/CreateTagFromWatermarkProcedureITCase.java [hunk 1]
```diff
@@ -80,30 +85,67 @@ public void testCreatTagsFromSnapshotsWatermark() throws Exception {
...
+                                                    + "`table` => 'default.T',"
+                                                    + "`tag` => 'tag2',"
+                                                    + "`watermark` => %s)",
+                                            watermark2 - 1)
+                                    .stream()
+                                    .map(Row::toString))
+                    .containsExactlyInAnyOrder(
+                            String.format("+I[tag2, 2, %s, %s]", commitTime2, watermark2));
+        } else {
+            createAction(
```
Comment (LinMingQiang) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/procedure/CreateTagFromWatermarkProcedureITCase.java:101]: You shouldn't add tests in ProcedureITCase.

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/procedure/CreateTagFromWatermarkProcedureITCase.java [hunk 2]
```diff
@@ -160,28 +202,72 @@ public void testCreatTagsFromTagsWatermark() throws Exception {
...
+                                                    + "`table` => 'default.T',"
+                                                    + "`tag` => 'tag2',"
+                                                    + "`watermark` => %s)",
+                                            tagsWatermark - 1)
+                                    .stream()
+                                    .map(Row::toString))
+                    .containsExactlyInAnyOrder(
+                            String.format("+I[tag2, 1, %s, %s]", tagsCommitTime, tagsWatermark));
+        } else {
+            createAction(
```
Comment (LinMingQiang) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/procedure/CreateTagFromWatermarkProcedureITCase.java:218]: ditto

File: paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/procedure/CreateTagFromWatermarkProcedureITCase.java [hunk 3]
```diff
@@ -160,28 +202,72 @@ public void testCreatTagsFromTagsWatermark() throws Exception {
...
+                            "--watermark",
+                            Long.toString(watermark2 - 1))
+                    .run();
+        }
+        assertThat(table.tagManager().tagExists("tag3")).isTrue();
+        assertThat(table.tagManager().taggedSnapshot("tag3").watermark()).isEqualTo(watermark2);
+        assertThat(table.tagManager().taggedSnapshot("tag3").timeMillis()).isEqualTo(commitTime2);
+    }
+
+    private <T extends ActionBase> T createAction(Class<T> clazz, String... args) {
```
Comment (LinMingQiang) [inline paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/procedure/CreateTagFromWatermarkProcedureITCase.java:267]: ditto

**Review by LinMingQiang — CHANGES_REQUESTED** [review]

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**